### PR TITLE
feat: add spark to substrait named table relation conversion with schema resolution

### PR DESCRIPTION
**Pull request by richtia**

### CODE REVIEW

File: src/gateway/server.py [hunk 1]
```diff
@@ -108,7 +110,9 @@ def ExecutePlan(
             case _:
                 raise ValueError(f'Unknown plan type: {request.plan}')
         _LOGGER.debug('  as Substrait: %s', substrait)
-        backend = find_backend(self._options.backend)
+        # backend = find_backend(self._options.backend)
+        backend = find_backend(BackendOptions(backend_engine.DUCKDB, True))
```
Comment (EpsilonPrime) [inline src/gateway/server.py:114]: You can set the backend to use on the client side by using setting spark-substrait-gateway.backend when setting up the spark session:

https://github.com/EpsilonPrime/spark-substrait-gateway/blob/main/src/gateway/tests/conftest.py#L46

File: src/gateway/server.py [hunk 2]
```diff
@@ -108,7 +110,9 @@ def ExecutePlan(
...
+        backend.register_tpch()
```
Comment (EpsilonPrime) [inline src/gateway/server.py:115]: The DuckDB backend already registers the TPC-H dataset, perhaps we should remove that from there and call it always for all backends:

https://github.com/EpsilonPrime/spark-substrait-gateway/blob/main/src/gateway/backends/duckdb_backend.py#L41

File: src/gateway/demo/client_demo.py [hunk 3]
```diff
@@ -33,9 +32,8 @@ def execute_query(spark_session: SparkSession) -> None:
 
     df_result.show()
 
-    sql_results = spark_session.sql(
-        'SELECT c_custkey, c_phone, c_mktsegment FROM customer LIMIT 5').collect()
-    print(sql_results)
+    name = 'customer'
```
Comment (EpsilonPrime) [inline src/gateway/demo/client_demo.py:35]: Revert the changes to this file so it's retains the same capabilities?

**Review by EpsilonPrime — APPROVED** [review]

File: src/gateway/demo/client_demo.py (hunk 3 above)
Comment (richtia) [reply src/gateway/demo/client_demo.py]: done
